package/watcher/watcher.go

```diff
@@ -162,11 +165,17 @@ func Watch(ctx context.Context, dir string, fn func(path string) error) error {
 					}
 
 				// Handle write events
-				case op&fsnotify.Write != 0:
+				// Currently, there is an issue with fsnotify which may register multiple events
+				// in a single change, so we have to ignore events that is triggered too
+				// closely together (details in #70)
+				// If there are multiple events in one file change, this only accept the first one.
+				case op&fsnotify.Write != 0 && time.Since(startTime).Milliseconds() > 1:
```

Comment:
This doesn't work on OSX, I guess because events occur soon than 1ms. I think the solution is closer to keeping a map of "last seen" files. If you last saw the file less than a millisecond ago, ignore it.

It's not perfect because that map will keep growing. Maybe we can use an LRU like https://github.com/hashicorp/golang-lru to make it bounded.

Reply:
The OSX test suite did correctly report this failure, so you can use that to test your solution.

Reply:
The problem here is time from this line of code
```golang
// Start timer
startTime := time.Now()
```
to the write handler is too short.
I made [a PR](https://github.com/livebud/bud/pull/88) to for the fix.

Reply:
The problem here is time from this line of code
```golang
// Start timer
startTime := time.Now()
```
to the write handler is too short.
I made [a PR](https://github.com/livebud/bud/pull/88) to for the fix.